src/nupic/math/SparseBinaryMatrix.hpp

```diff
@@ -2002,7 +2002,7 @@ namespace nupic {
 	  << "Invalid index: " << *it
 	  << " - Should be >= 0 and < number of columns:" << nCols();
       
-      for (size_type i = 1; i < (size_type)(end - begin); ++i)
+      for (auto i = 1; i < (end - begin); ++i)
```

Comment:
> 1 can have many types so I would not use auto here.

@scottpurdy, I tried several times to build using `size_type` for this loop, however this error: 

`error C2593: 'operator +' is ambiguous`

arise 3 times for this line:

`NTA_ASSERT(*(begin + i - 1) < *(begin + i))`

:disappointed: